app/ts/components/simulationExplaining/customExplainers/GovernanceVoteVisualizer.tsx

```diff
@@ -67,7 +86,13 @@ export function GovernanceVoteVisualizer(param: TransactionImportanceBlockParams
 			<div class = 'notification dashed-notification'>
 				{ simulateGovernanceContractExecutionReply !== undefined ? <>
 					<div style = 'display: grid; grid-template-rows: max-content' >
-						{ 'error' in simulateGovernanceContractExecutionReply.data ? <ErrorComponent warning = { false } text = { simulateGovernanceContractExecutionReply.data.error.message }/> : <>
+						{ 'error' in simulateGovernanceContractExecutionReply.data ? <>
+							{ simulateGovernanceContractExecutionReply.data.error.type === 'MissingAbi' ? <MissingAbi
+								errorMessage = { simulateGovernanceContractExecutionReply.data.error.message }
+								addressBookEntry = { simulateGovernanceContractExecutionReply.data.error.addressBookEntry }
+								renameAddressCallBack = { param.renameAddressCallBack }
+							/> : <ErrorComponent warning = { false } text = { simulateGovernanceContractExecutionReply.data.error.message }/> }
+						</> : <>
```

Comment:
Similar comment as above.  Anytime I find myself doing ternaries inside of my TSX blocks I find it valuable to extract the code out into a function-local component and put the ternary in there.  I find this makes the TSX much more readable as you just have a bunch of named components in a hierarchy rather than a bunch of code.  You can also have multiple layers of named function-local components if you have a situation like this where you have a conditional inside a conditional.